src/mpid/ch4/shm/include/shm.h

```diff
@@ -14,7 +14,13 @@
 
 #include <mpidimpl.h>
 
+enum {
+    MPIDI_SHM_GO_XPMEM = 0,
+    MPIDI_SHM_GO_POSIX
+};
+
 #define MPIDI_MAX_SHM_STRING_LEN 64
+#define MPIDI_SHM_FALLBACK_THRESHOLD (16 * 1024 - sizeof(uint64_t) - 2 * sizeof(int))
 
```

Comment:
Need better name for it. It is used as starting message size of XPMEM, or max message size of POSIX eager. We may also add other thresholds for PiP or CMA in future.